course-material/interactive-queries-and-parameterization/make-your-queries-interactive.md

```diff
@@ -1,14 +1,404 @@
+<!-- #region -->
+<details>
+<summary>Answers</summary>
+
+<b>Hint</b> Did you know we can also create animated sliders for integer data types? This question requires exactly that! See the documentation [here](https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20List.html#play-animation-widget) for more details.
+
+We start off by initializing a variable called `play` and assigning it to the `Play` Animation widget with the required <b>integer</b> arguments.
+
+```{code-cell} ipython3
+play = widgets.Play(
+    value=0,
+    min=0,
+    max=2000,
+    step=500,
+    interval=1000,  # time interval in milliseconds
+    description="Press play",
+    disabled=False,
+)
+```
+
+For the `ToggleButton` widget, we initialize a variable called `housing_toggle` and assign it to the `ToggleButton` widget with the required arguments.
+
+```{code-cell} ipython3
+housing_toggle = widgets.ToggleButtons(
+    options=["yes", "no"],
+    description="Housing:",
+    disabled=False,
+)
+```
+
+Before calling `--interact`, we need to add UI's for the `Play` and `IntSlider` widgets. This is attained with both `jslink()` and `HBox` `ipywidgets` methods. We then use the `--interact` argument to create the UI's. In the `WHERE` clause, because we want an <b>unbounded</b> slider, we use the `>=` operator for `duration`. To limit the number of rows to only 10, we <b>do not need a basic slider</b>.
+
+```{code-cell} ipython3
+%%sql --interact play --interact housing_toggle 
+SELECT * FROM bank
+WHERE duration >= {{play}} AND                                                         
+housing == '{{housing_toggle}}' 
+LIMIT 10
+```
+
+```{important}
+Because we set the minimum value and initial value upon rendering to 0, the maximum value to 2000, and a step size of 500, the table will change or "blink" four times upon pressing the "play" button. Normally, an `IntSlider` is recommended to be added next to the `Play` widget; however, JupySQL does not support this at the moment.
+```
+
++++
+
+</details>
+<!-- #endregion -->
+
+<!-- #region -->
+
+## Wrapping Up
+
+In this section, we learnt about how to create widgets to make our SQL queries interactive, helping us gain a better intuition of the EDA process! To summarize:
+
+- Numeric widgets, although commonly used as sliders, can also be used in text boxes or dropdowns, for example. This is useful when you want to specify a value.
+
+- For categorical or text data, we can use either Selection or Boolean widgets. Some examples include radio buttons, toggle buttons, and dropdowns.
```

Comment:
Add summary of the difference between % and %%

also that you can chain -- options like --save and --interact and that furthermore you can combine sql queries with macros and widgets